rero_ils/modules/selfcheck/api.py

```diff
@@ -0,0 +1,143 @@
+from ..patron_transactions.api import PatronTransaction
+from ..patrons.api import Patron
+
+
+def selfcheck_login(login, password):
+    """Selfcheck login handler."""
+    if authorize_selfckeck_user(login, password):
+        staffer = Patron.get_patron_by_email(login)
+        return {
+            'authenticated': staffer.is_librarian,
+            'institution_id': staffer.get_organisation().get('code'),
+            'library_name': Library.get_record_by_pid(
+                staffer.library_pid).get('name')
+        }
+
+
+def validate_patron_account(barcode=None):
+    """Validate patron account handler."""
+    patron = Patron.get_patron_by_barcode(barcode)
+    if patron and patron.is_patron:
+        return True
+
+
+def authorize_patron(barcode, password):
+    """Authorize patron handler."""
+    patron = Patron.get_patron_by_barcode(barcode)
+    if patron and patron.is_patron:
+        return authorize_selfckeck_user(patron.get('email'), password)
+
+
+def system_status(login, *kwargs):
+    """Selfcheck system status handler."""
+    staffer = Patron.get_patron_by_email(login)
+    return {
+        'institution_id': staffer.get_organisation().get('code')
+    }
+
+
+def enable_patron(barcode, **kwargs):
+    """Selfcheck enable patron handler."""
+    # check if invenio_sip2 module is present
+    if check_sip2_module():
+        from invenio_sip2.models import SelfcheckPatronStatus
+
+        patron = Patron.get_patron_by_barcode(barcode)
+        patron_status = SelfcheckPatronStatus()
+        # TODO: add PatronStatusType according account
+        #  e.g.:
+        #   patron_status.add_patron_status_type(
+        #   SelfcheckPatronStatusTypes.CARD_REPORTED_LOST)
+        return {
+            'patron_status': patron_status,
+            'language': patron.get('communication_language', 'und'),
+            'institution_id': patron.get_organisation().get('code'),
+            'patron_id': patron.get('barcode'),
+            'patron_name': patron.formatted_name
+        }
+
+
+def patron_information(barcode, **kwargs):
+    """Get patron information."""
+    # check if invenio_sip2 module is present
+    if check_sip2_module():
+        from invenio_sip2.models import SelfcheckPatronInformation
+
+        patron = Patron.get_patron_by_barcode(barcode)
+        patron_account_information = SelfcheckPatronInformation(
+            patron_id=barcode,
+            patron_name=patron.formatted_name,
+            patron_email=patron.get('email'),
+            patron_phone=patron.get('phone'),
+            patron_address=format_patron_address(patron),
+            institution_id=patron.get_organisation().get('code'),
+            language=patron.get('communication_language', 'und'),
+            currency_type=patron.get_organisation().get('default_currency')
+        )
+
+        loans = get_loans_by_patron_pid(patron.pid)
+        for loan in loans:
+            item = Item.get_record_by_pid(loan.get('item_pid'))
```

Comment:
`item = Item.get_record_by_pid(loan.get('item_pid'))``
replace by
`item = Item.get_record_by_pid(loan.item_pid)`
because loan.get('item_pid') is an object

Reply:
ok